dist-m39/src/main/provisio/maven-distro.xml

```diff
@@ -0,0 +1,93 @@
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+-->
+<assembly>
+
+    <artifactSet to="/">
+        <artifact id="org.apache.maven:apache-maven:tar.gz:bin:${maven3.version}">
+            <unpack useRoot="false"
+                    excludes="conf/logging/*,lib/maven-slf4j-provider*" />
+        </artifact>
+    </artifactSet>
+
+    <artifactSet to="/lib">
+        <artifact id="ch.qos.logback:logback-classic">
+            <exclusion id="*:*"/>
+        </artifact>
+        <artifact id="ch.qos.logback:logback-core">
+            <exclusion id="*:*"/>
+        </artifact>
+    </artifactSet>
+
+    <artifactSet to="/lib/ext">
+        <artifact id="org.apache.maven.daemon:mvnd-daemon:${project.version}">
+            <exclusion id="*:*"/>
+        </artifact>
+        <artifact id="org.apache.maven.daemon:mvnd-daemon-m39:${project.version}">
+            <exclusion id="*:*"/>
+        </artifact>
+        <artifact id="org.apache.maven.daemon:mvnd-client:${project.version}">
+            <exclusion id="*:*"/>
+        </artifact>
+        <artifact id="org.apache.maven.daemon:mvnd-common:${project.version}">
+            <exclusion id="*:*"/>
+        </artifact>
+        <artifact id="org.apache.maven.daemon:mvnd-agent:${project.version}">
+            <exclusion id="*:*"/>
+        </artifact>
+        <artifact id="org.apache.maven.daemon:mvnd-helper-agent:${project.version}">
+            <exclusion id="*:*"/>
+        </artifact>
+        <artifact id="org.apache.maven.daemon:mvnd-native:${project.version}">
+            <exclusion id="*:*"/>
+        </artifact>
+        <artifact id="org.codehaus.plexus:plexus-interactivity-api">
+            <exclusion id="*:*"/>
+        </artifact>
+        <artifact id="org.jline:jline-terminal">
+            <exclusion id="*:*"/>
+        </artifact>
+        <artifact id="org.jline:jline-terminal-jansi">
+            <exclusion id="*:*"/>
+        </artifact>
+    </artifactSet>
+
+    <fileSet to="/">
+        <directory path="${basedir}/../dist/src/main/distro"/>
+        <directory path="${basedir}/..">
+            <include>NOTICE.txt</include>
+            <include>LICENSE.txt</include>
+            <include>README.adoc</include>
+        </directory>
+    </fileSet>
+    <fileSet to="/bin">
+        <directory path="${basedir}/../client/target">
+            <include>mvnd</include>
+            <include>mvnd.exe</include>
+        </directory>
+        <directory path="${basedir}/../dist/src/main/resources">
+            <include>platform-${os.detected.name}-${os.detected.arch}</include>
+        </directory>
+    </fileSet>
+
+    <archive name="maven-${project.version}-mvnd-mvn39-${os.detected.name}-${os.detected.arch}.zip"
+             executable="**/bin/mvnd"/>
+
+    <archive name="maven-${project.version}-mvnd-mvn39-${os.detected.name}-${os.detected.arch}.tar.gz"
+             executable="**/bin/mvnd"/>
```

Comment:
Just thinking out loud - this produces a file named e.g. `maven-1.0.0-m5-SNAPSHOT-mvnd-mvn39-linux-amd64.tar.gz` which to _me_ sounds like this is somewhat related to Maven version `1.0.0-m5-SNAPSHOT`, which is not really the case. This is the version of the Maven daemon. Was there a specific reason to change the naming scheme compared to the older branches (e.g. `maven-mvnd-0.10.0-SNAPSHOT-linux-amd64.tar.gz`)? Would it make sense to go back to `maven-mvnd-1.0.0-m5-SNAPSHOT-mvn39-linux-amd64.tar.gz`?